### ISSUE TITLE
Ability to modify JSON fields in weblog request

### ISSUE THREAD

**Issue by ODiogoSilva** [nextflow-io/nextflow#881]
Currently, the weblog functionality of nextflow sends a POST request with a fixed set of JSON keys/field (for instance `runName`, `runId`, etc). However, it would be useful to have the ability to change/specify the keys that should be sent. The way these fields are specified could be very similar to the way it is done for the `trace` scope using the `fields` option.

@sven1103

**Comment by ODiogoSilva** [comment]
The fields in the trace file also seem to ignore the fields specified in the `trace` configuration in nextflow.config. It would be nice if this could be controlled in the same way as the trace file.

**Comment by ewels** [comment]
As this is a small machine-readable data packet, any reason not to just put _all_ fields in here every time? Then the end user can choose what to use. Would mean that downstream tools could be written without having to require pipeline users to do any configuration.

**Comment by ODiogoSilva** [comment]
Yes, it's true, but the same can be said about the `trace` information. By enabling it, it writes the default fields, but then you can also modify which fields get saved. 

The packets are small individually, but it may take a toll after a few tens of thousand requests (multiplied by the process_submitted, process_started and process_completed requests for each process). For instance, 10k processes generating packets of just 10Kb would generate about 300Mb of network (the size seems to be largely dependent on the size of the script file passing through). I understand that this would only be a real issue for larger workflows, though.

Basically, if we could have some control over what is sent, it would be great, but if not, we can always do as @ewels suggested.

**Comment by ewels** [comment]
Fair enough, all good points 😀
